### PR TITLE
HTML: Test U+0000 NULL in textarea's value/defaultValue/textContent

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/w3c/web-platform-tests/pull/5272#issuecomment-307257151

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/241064118)

# Firefox (nightly) #
Testing web-platform-tests at revision bd94c0316910e52cb2999a4eb33a23ec3f9c7163
Using browser at version BuildID 20170608100220; SourceStamp 7efda263a842e60cd0cc00b3c4a7058c65590702
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6197/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                            Subtest                                             | Results |                                                               Messages                                                              |
|------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                                | OK      |                                                                                                                                     |
| `defaultValue and value are the empty string by default`                                       | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by setting textContent`                                   | PASS    |                                                                                                                                     |
| `defaultValue and value are affected by textContent in combination with appending a text node` | PASS    |                                                                                                                                     |
| `defaultValue and value treat CRLF differently`                                                | FAIL    | `assert_equals: The value property should normalize CRLF and CR to LF expected "foo\nbar\nbaz\nqux" but got "foo\r\nbar\rbaz\nqux"` |
| `tests for the value setter`                                                                   | PASS    |                                                                                                                                     |
| `tests for U+0000 NULL`                                                                        | PASS    |                                                                                                                                     |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/241064119)

# Sauce (safari) #
Testing web-platform-tests at revision bd94c0316910e52cb2999a4eb33a23ec3f9c7163
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6197/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html">/html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html</a></summary>


|                                            Subtest                                             | Results | Messages |
|------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                | OK      |          |
| `defaultValue and value are the empty string by default`                                       | PASS    |          |
| `defaultValue and value are affected by setting textContent`                                   | PASS    |          |
| `defaultValue and value are affected by textContent in combination with appending a text node` | PASS    |          |
| `defaultValue and value treat CRLF differently`                                                | PASS    |          |
| `tests for the value setter`                                                                   | PASS    |          |
| `tests for U+0000 NULL`                                                                        | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/241064125)

# Chrome (unstable) #
Testing web-platform-tests at revision bd94c0316910e52cb2999a4eb33a23ec3f9c7163

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/241064126)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision bd94c0316910e52cb2999a4eb33a23ec3f9c7163

**Comment by zcorpan** [comment]
(Reusing https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/12263595/ instead of filing a new issue for this.)

**Review by cvrebert — APPROVED** [review]

File: html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html [hunk 1]
```diff
@@ -68,4 +68,25 @@
   assert_equals(textarea.value, "", "setting the value property to null should result in an empty string");
 
 }, "tests for the value setter");
+
+test(() => {
+
+  const textarea = document.createElement("textarea");
+  textarea.defaultValue = "foo\0";
```
Comment (zcorpan) [inline html/semantics/forms/the-textarea-element/value-defaultValue-textContent.html]: I tried to be consistent with the style this file was already using. ^_^

**Pull request merged**